### PR TITLE
Block AD in Large preview

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

**Comment by revonateB0T** [comment]
1.Recompose has some problem
2.Try size original to detect
I will refine it later.

**Comment by FooIbar** [comment]
The false negative rate is too high to justify shipping it.
Less than 1/4 images from [the test gallery](https://e-hentai.org/g/2455300/2945cac7a8) were detected.

**Comment by revonateB0T** [comment]
@FooIbar Then I have no idea, any further possibility to improve it?

**Comment by revonateB0T** [comment]
@revonateB0T If we could improve it later, we just land it as this MR doesn't touch rust code anyway.

**Comment by FooIbar** [comment]
> @FooIbar Then I have no idea, any further possibility to improve it?

I think detecting thumbnails is a dead end.
The URL of the large preview contains the original image's hash, maybe we can use that if we want to block ad thumbnails.

File: app/src/main/kotlin/com/hippo/ehviewer/client/data/GalleryInfo.kt [hunk 1]
```diff
@@ -142,5 +144,4 @@ fun GalleryInfo.findBaseInfo(): BaseGalleryInfo = when (this) {
 
 fun GalleryInfo.asGalleryDetail(): GalleryDetail? = this as? GalleryDetail
 
-val GalleryInfo.hasAds
-    get() = simpleTags?.any { "extraneous ads" in it } ?: false
+fun detectAds(index: Int, size: Int) = index > size - 10 && Settings.stripExtraneousAds.value
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/client/data/GalleryInfo.kt:147]: We can merge without enabling it If you insist, let this always return false.
Also nit: this file is not a well place for this function.

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/client/data/GalleryInfo.kt:147]: Why we not enabling it?
It can block some ADs at least and do no harm.

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/client/data/GalleryInfo.kt]: Done

**Review by FooIbar — APPROVED** [review]

**Pull request merged**